controllers/toolchainstatus/toolchainstatus_controller_test.go

```diff
@@ -1785,6 +2031,15 @@ func proxyRoute() *routev1.Route {
 	}
 }
 
+func proxyRouteInvalid() *routev1.Route {
+	return &routev1.Route{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "api",
+			Namespace: test.HostOperatorNs,
+		},
+	}
+}
```

Comment:
to trigger the url parsing error you need to specify an invalid URL as well. A blank one is not enough. 
```suggestion
	return &routev1.Route{
		ObjectMeta: metav1.ObjectMeta{
			Name:      "api",
			Namespace: test.HostOperatorNs,
		},
		Spec: routev1.RouteSpec{
			Host: ":invalidhost",
			Port: &routev1.RoutePort{
				TargetPort: intstr.FromInt(8081),
			},
			TLS: &routev1.TLSConfig{
				Termination: routev1.TLSTerminationEdge,
			},
		},
	}
```

Reply:
Yeah this works, I was trying to set the Host to `invalidhost` instead of `:invalidhost`, hence the error was not reachable .